### ISSUE TITLE
Places incorrect paths in extracted source maps for input that has relative sources

### ISSUE THREAD

**Issue by rjgotten** [webpack-contrib/css-loader#1225]
* Operating System: Windows 10
* Node Version: 10.15.3
* NPM Version: 6.9.0
* webpack Version: 4.31.0
* mini-css-extract-plugin Version: 0.7.0

### Expected Behavior

When offered compiled CSS content backed by a source map where `sources` are relative URLs, extracts a `.css` file with a source map that has correct path information. That is: the `sources` are resolved against `this.context` of the compiled CSS content, producing the correct source locations.

### Actual Behavior

When offered compiled CSS content backed by a source map where `sources` are relative URLs, extracts a `css` file with a source map that has incorrect path information. The `sources` seem to be output as-is to be interpreted by debuggers and dev-tools as relative to the extracted CSS file. This produces incorrect source locations.

### Code

N/A

### How Do We Reproduce?

Minimal repro case:

https://github.com/NetMatch/mini-css-extract-plugin-testcase

**Comment by evilebottnawi** [comment]
@rjgotten Do you have ideas how we can achieve this?

**Comment by rjgotten** [comment]
Maybe just cop out? Introduce a rule that says the first candidate matching a given predicate test is passed on.
Or introduce a system which favors one extension type over the other?

It's situational, I think. So probably a user specified predicate/test regex or function is the better deal.

**Comment by evilebottnawi** [comment]
@rjgotten We can't rewrite source maps, PostCSS does it, we just provide `from`/`to` and previously source map

**Comment by rjgotten** [comment]
Ah sorry, about the confusion there.

I didn't mean "rewrite the source-map" as in creating a chained source map, which indeed is what PostCSS should be doing.
I meant to rewrite the `sources` attribute of the source-map that is passed to PostCSS, so it only shows _one_ source to PostCSS.

Thinking further though: why is the presence of multiple `sources` even an issue for the `from` attribute?
The PostCSS `from` and `to` options are the input and output CSS. And iirc the only connection `from` has to source-maps is that the file name provided there will be used as the content of a source-map's `sources` in the event that PostCSS starts a _new_ source map - i.e. when it is *not* chaining off of an existing one. In case of an incoming source-map that already has multiple sources, that case obviously doesn't - or at least: shouldn't - apply.

**Comment by evilebottnawi** [comment]
We can pass to `from` first `source` (`map.sources[0]`) if we have it in source map, so we can achieve this

**Comment by rjgotten** [comment]
Oh; right. _Now_ I get what the problem is.
For compatibility's sake, you need to set a name for the source, even though you're passing it in programmatically and not from disk and even though you're chaining off of an existing source map. (Yeah; that took a moment to click with me... sorry.)

Sure; in that case just going with whatever is listed as the first entry in `map.sources` should work.
Heck, just passing in any arbitrary string _should_ work, I guess, because it should never end up in the output source-map. The output source-map should be propagating the original sources from the input source-map, after all.

**Comment by evilebottnawi** [comment]
Yep, I am worried that some of the tools may be generating incorrect invalid order of `sources`, but I think this is not our problem

**Comment by nineninesevenfour** [comment]
Hi! I am new to webpack and css-loader, but I want to share my observations and hope it helps, but if its not relevant for this issue, please ignore it. thx. :wink:
While going through the "Asset Management" tutorial (and playing around with source maps) I ran webpack with the `--stats verbose` option and get the following output:
```
ModuleConcatenation bailout: Cannot concat with ./node_modules/css-loader/dist/cjs.js??ruleSet[1].rules[0].use[1]!./src/style.css: Module uses module.id
```
On [Stackoverflow](https://stackoverflow.com/questions/45384170/how-to-fix-modules-with-moduleconcatenation-bailout-module-is-not-an-ecmascrip) I can read the following about it:
> ...Webpack receives the CommonJS modules, but the ModuleConcatenationPlugin relies on ES modules....

I found no option to change the internal behaviour of css-loader from CJS towards ES6 modules.
Funnily Firefox does not display me the `style.css` at all, while in Safari I get two style.css: one is the js-output of `css-loader` and one is the js-output of `style-loader`.  Obviously somehow the css source map resolves to an intermediate result instead of the original one. And I suppose it might be due to a simple incompatibilty with the bundled ModuleConcatenationPlugin.

Btw. I removed `babel-loader` from the `webpack.config.js` to get sure that is not intervening that process.
